src/TestFramework/PhpUnit/CommandLine/ArgumentsAndOptionsBuilder.php

```diff
@@ -57,21 +60,34 @@ final class ArgumentsAndOptionsBuilder implements CommandLineArgumentsAndOptions
 {
     private const MAX_EXPLODE_PARTS = 2;
 
-    public function __construct(private readonly bool $executeOnlyCoveringTestCases)
-    {
+    /**
+     * @param list<SplFileInfo> $filteredSourceFilesToMutate
+     */
+    public function __construct(
+        private readonly bool $executeOnlyCoveringTestCases,
+        private readonly array $filteredSourceFilesToMutate,
+        private readonly ?string $mapSourceClassToTestStrategy,
+    ) {
     }
 
+    /**
+     * @return list<string>
+     */
     public function buildForInitialTestsRun(string $configPath, string $extraOptions): array
     {
-        $options = [
-            '--configuration',
-            $configPath,
-        ];
+        $options = $this->prepareArguments($configPath, $extraOptions);
 
-        if ($extraOptions !== '') {
-            $options = array_merge(
-                $options,
-                array_map(static fn ($option): string => '--' . $option, explode(' --', ltrim($extraOptions, '-')))
+        if ($this->filteredSourceFilesToMutate !== []
+            && $this->mapSourceClassToTestStrategy !== null
+            && !in_array('--filter', $options, true)) {
+            $options[] = '--filter';
+
+            $options[] = implode(
+                '|',
+                array_map(
+                    static fn (SplFileInfo $sourceFile): string => sprintf('%sTest', $sourceFile->getBasename('.' . $sourceFile->getExtension())),
+                    $this->filteredSourceFilesToMutate,
+                )
```

Comment:
These 10 lines actually the real implementation of the feature that needs attention.

as I wrote in the description, this is a `simple` strategy that maps `SourceClass` to `SourceClassTest`